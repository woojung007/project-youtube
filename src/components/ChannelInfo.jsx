import { useQuery } from '@tanstack/react-query';
import { useYoutubeApi } from 'context/YoutubeApiContext';
import React from 'react';

export default function ChannelInfo({ id, name }) {
    const { youtube } = useYoutubeApi();
    const { data: url } = useQuery(['channel', id], () => youtube.channelImageUrl(id));

    return (
        <div className='flex items-center my-4 mb-8'>
            {url && <img className='w-10 h-10 rounded-full' src={url} alt={name} />}
            <p className='ml-2 text-lg font-medium'>{name}</p>
        </div>
    );
}
