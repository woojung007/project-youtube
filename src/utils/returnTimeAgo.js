import moment from '../../node_modules/moment/moment';

export const returnTimeAgo = (date) => {
    const now = new Date();

    return moment(date).from(moment(now));
};
