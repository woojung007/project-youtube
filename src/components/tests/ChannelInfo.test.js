/* eslint-disable testing-library/prefer-find-by */
import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import ChannelInfo from 'components/ChannelInfo';
import { Route } from 'react-router-dom';
import { withAllContexts, withRouter } from 'tests/utils';

describe('ChannelInfo', () => {
    // class(object instance)를 흉내내는 가짜 object
    const fakeYoutube = {
        channelImageUrl: jest.fn(),
    };

    afterEach(() => {
        fakeYoutube.channelImageUrl.mockReset();
    });

    /**
     * 1. snapshot(img, p)
     * 2. url X
     * (3. url O)
     */

    it('renders correctly', async () => {
        fakeYoutube.channelImageUrl.mockImplementation(() => 'url');

        const { asFragment } = render(
            withAllContexts(
                withRouter(<Route path='/' element={<ChannelInfo id='id' name='channel' />} />),
                fakeYoutube,
            ),
        );

        await waitFor(() => expect(screen.getByRole('img')));

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders without URL', () => {
        fakeYoutube.channelImageUrl.mockImplementation(() => {
            throw new Error('error');
        });

        render(
            withAllContexts(withRouter(<Route path='/' element={<ChannelInfo id='id' name='channel' />} />)),
            fakeYoutube,
        );

        expect(screen.queryByRole('img')).toBeNull();
    });

    it('renders with URL', async () => {
        fakeYoutube.channelImageUrl.mockImplementation(() => 'url');

        render(
            withAllContexts(
                withRouter(<Route path='/' element={<ChannelInfo id='id' name='channel' />} />),
                fakeYoutube,
            ),
        );

        await waitFor(() => expect(screen.getByRole('img')).toBeInTheDocument());
    });
});
