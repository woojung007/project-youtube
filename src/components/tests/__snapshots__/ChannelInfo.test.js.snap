// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center my-4 mb-8"
  >
    <img
      alt="channel"
      class="w-10 h-10 rounded-full"
      src="url"
    />
    <p
      class="ml-2 text-lg font-medium"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
